security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -205,6 +215,32 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 	}
 
 	conIDresourceNamePrefix := cacheLogPrefix(connectionID, resourceName)
+
+	// When there are existing root certificates, or private key and certificate under
+	// a well known path, they are used in the SDS response.
+	// In the current implementation, the file update events are not handled and if
+	// the files are updated, a user may restart Envoy to pick up the updated files.
+	// TODO (lei-tang): if updating files are supported, add a file watcher for
+	// the files under the well known path.
+	sdsFromFile := false
+	var err error
+	if sc.rootCertificateExists() && connKey.ResourceName == RootCertReqResourceName {
+		sdsFromFile = true
+		ns, err = sc.generateRootCertFromExistingFile(token, connKey)
+	} else if sc.keyCertificateExists() && connKey.ResourceName == WorkloadKeyCertResourceName {
+		sdsFromFile = true
+		ns, err = sc.generateKeyCertFromExistingFiles(token, connKey)
+	}
+	if sdsFromFile {
+		if err != nil {
+			cacheLog.Errorf("%s failed to generate secret for proxy: %v",
```

Comment:
"%s failed to generate secret for proxy: %v, by loading from local files",

Reply:
Done.